### PR TITLE
feat: Add Converse component and improve output extraction

### PR DESCRIPTION
**Pull request by Cristhianzl**
This pull request introduces significant improvements to Amazon Bedrock model support and output text extraction, with a focus on compatibility, robustness, and expanded testing. The most notable changes are the addition of a new Amazon Bedrock Converse component, a major rewrite of the `_extract_output_text` function to handle diverse response formats gracefully, and comprehensive unit tests to ensure reliability across edge cases and backward compatibility.

**Amazon Bedrock Model Components:**

* Added new `AmazonBedrockConverseComponent` in `amazon_bedrock_converse.py`, supporting the modern Converse API for improved conversation handling, more flexible model parameters, and robust error handling for unsupported models or parameters.
* Updated the legacy `AmazonBedrockComponent` description to recommend the new Converse component for most users, clarifying its intended use and compatibility.

**Output Text Extraction Improvements:**

* Refactored the `_extract_output_text` function in `events.py` to handle a wide variety of output formats, including ChatBedrockConverse responses, Anthropic-style outputs, and backward compatibility scenarios. The function now gracefully handles index-only items, tool use, partial JSON, mixed types, and multiple text segments, returning an empty string for non-text items.

**Testing Enhancements:**

* Added a comprehensive suite of unit tests for `_extract_output_text` in `test_agent_events.py`, covering all major formats, edge cases, and backward compatibility scenarios, including specific cases that previously caused errors with ChatBedrockConverse.
* Imported the `_extract_output_text` function in the test file to enable these new tests.

## Summary by CodeRabbit

- New Features
  - Added Amazon Bedrock Converse component with model selection, AWS credentials, region/endpoint support, generation controls, streaming toggle, and extra request fields, plus clearer initialization errors.
- Bug Fixes
  - Improved reliability when extracting text from varied provider outputs and streaming chunks, avoiding tool-call and partial-JSON noise.
- Documentation
  - Updated legacy Amazon Bedrock component description to recommend the Converse API for improved compatibility and features.
- Tests
  - Added extensive test coverage for output parsing across diverse formats and edge cases.

### CODE REVIEW

File: src/backend/tests/unit/base/mcp/test_mcp_util.py [hunk 1]
```diff
@@ -251,6 +251,366 @@ async def test_sse_client_header_storage(self):
         assert sse_client._connection_params["headers"] == test_headers
 
 
+class TestFieldNameConversion:
```
Comment (edwinjosechittilappilly) [inline src/backend/tests/unit/base/mcp/test_mcp_util.py:254]: Should we have this change in in this PR?

**Comment by Cristhianzl** [comment]
please follow up this PR: #9901
